lcftrans/src/translation.cpp

```diff
@@ -141,33 +181,30 @@ static void parseEvents(Translation& t, lcf::StringView parent_name, U& root, co
 		info.clear();
 	};
 
-	// Read common events
-	lcf::rpg::ForEachString(root, nullptr, [&](lcf::Context<lcf::DBString>& ctx) {
-		if (!ctx.parent || !ctx.parent->parent) {
+	// Read events
+	lcf::rpg::ForEachString(root, [&](const auto& val, const auto& ctx) {
+		if (!isEventCommandString(ctx) || !std::is_same<decltype(ctx.parent->obj), std::add_pointer_t<ParentType>>::value) {
```

Comment:
I use the ``ParentType`` to type filter because I must do a distinction between common event and battle event (they both have the same root ``lcf::Data::data`` (the database).

Is this a good way to do the filtering? My old check was a string comparison but imo this is more robust.

Reply:
You should consider taking this whole lambda and dispatching to another function with a null base case and an implemented EventCommand case.

The problem here is that because this lambda is compiled for every possible type, you need all these wrappers to make it compile. So for example `rpg::Actor`m the compiler will run through this code and compile all of it and complain if any of it doesn't compile.

By dispatching early, you get out of "generic" world and into "EventCommand" world. As soon as you do that, you can just directly get the code, indent, etc.. without all the wrappers, making your code cleaner and more readable.

For the parent filtering, the way it's written yes you need to do that if statement.

However, another way if you already have the ParentCtx object, that has the struct, so instead of calling ForEachString on the database root, call ForEachString on the CommonEvent / Troop parent directly. Then you need don't need the is_same stuff.

Reply:
because I needed lots of state I created a small class with two parse methods: One for evt commands, one "catch-all".
Helped to simplify the code alot :+1: